From bunk@stusta.de Fri Mar 10 14:25:09 2006
Date: Fri, 10 Mar 2006 23:25:06 +0100
From: Adrian Bunk <bunk@stusta.de>
To: gregkh@suse.de
Subject: USB: vicam.c: fix a NULL pointer dereference
Message-ID: <20060310222506.GA21864@stusta.de>
Content-Disposition: inline

This patch fixes a NULL pointer dereference spotted by the Coverity
checker.


Signed-off-by: Adrian Bunk <bunk@stusta.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/media/vicam.c |    1 +
 1 file changed, 1 insertion(+)

--- gregkh-2.6.orig/drivers/usb/media/vicam.c
+++ gregkh-2.6/drivers/usb/media/vicam.c
@@ -764,6 +764,7 @@ vicam_open(struct inode *inode, struct f
 	if (!cam) {
 		printk(KERN_ERR
 		       "vicam video_device improperly initialized");
+		return -EINVAL;
 	}
 
 	/* the videodev_lock held above us protects us from
